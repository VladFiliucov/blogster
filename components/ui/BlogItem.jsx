import React from 'react';
import PropTypes from 'prop-types';

import Image from './Image';
import TextBox from './TextBox';
import PostDetails from './PostDetails';
import Like from './Like';

const BlogItem = (props) => (
  <div>
    <Image image={props.post.image} />
    <TextBox>{props.post.text}</TextBox>
    <PostDetails details={props.post.details} />
    <Like likes={props.post.details.likes} />
  </div>
)

BlogItem.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.number,
    text: PropTypes.string,
    details: PropTypes.shape({
      author: PropTypes.string,
      created_at: PropTypes.string,
      updated_at: PropTypes.string,
      likes: PropTypes.number
    }),
    image: PropTypes.shape({
      src: PropTypes.string,
      alt: PropTypes.string,
      style: PropTypes.shape({
        width: PropTypes.number,
        height: PropTypes.number
      })
    })
  })
}

export default BlogItem;
