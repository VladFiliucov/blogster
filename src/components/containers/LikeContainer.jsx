import { connect } from 'react-redux';
import { likePost } from 'actions/Posts';

import Like from 'components/ui/Like';

function mapDispatchToProps(dispatch) {
  return {
    likePost: (posts, postId) => dispatch(likePost(posts, postId))
  };
}

export default connect(null, mapDispatchToProps)(Like);
