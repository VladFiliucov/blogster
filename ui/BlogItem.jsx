import React from 'react';

import Image from './Image';
import TextBox from './TextBox';

const style = {width: 200, height: 200}
const imageURL = "https://s-media-cache-ak0.pinimg.com/736x/8b/d1/b8/8bd1b8b02e9cb633adb1d789dd4588c6.jpg"
const altText= "Guy rides a co...grown chicken"
const text = "Lorem ipsum could be here"

const BlogItem = () => (
  <Image
    imageAddress={imageURL}
    style={style}
    altText={altText}
  />
  <TextBox text={text} />
)

export default BlogItem;
