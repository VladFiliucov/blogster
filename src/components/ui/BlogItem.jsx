import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Image from './Image';
import TextBox from './TextBox';
import PostDetails from './PostDetails';
import LikeContainer from 'components/containers/LikeContainer';

import 'components/styles/blog/BlogItem.css'

const BlogItem = ({ post }) => (
  <div className="wrapper card radius shadowDepth1">
    { post &&
        <div>
          <Link  to={post.postUrl} >
            <Image image={post.image} />
          </Link>
          <div className="card__content card__padding">
            <PostDetails details={post.details} />
            <TextBox>{post.text}</TextBox>
            <LikeContainer postId={post.id} />
          </div>
        </div>
    }
  </div>
);

BlogItem.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.number,
    text: PropTypes.string,
    details: PostDetails.propTypes.details,
    image: Image.propTypes.image
  })
};

export default BlogItem;
