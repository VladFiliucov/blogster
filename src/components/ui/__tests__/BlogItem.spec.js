import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import BlogItem from '../BlogItem';
import LikeContainer from 'components/containers/LikeContainer';
import 'components/styles/blog/BlogItem.css'

describe('BlogItem', () => {
  it('renders withour crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BlogItem />, div);
  });
});

