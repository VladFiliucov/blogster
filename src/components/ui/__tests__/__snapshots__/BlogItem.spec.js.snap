// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem render should render usual blog item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="wrapper card radius shadowDepth1"
>
    <div>
        <Link
            replace={false}
            to="localhost:9090/posts/4"
        >
            <Image
                image={
                    Object {
                        "alt": "Who cares",
                        "src": "/pictures/picture1.jpg",
                        "style": Object {
                          "height": 200,
                          "width": 200,
                        },
                      }
                }
            />
        </Link>
        <div
            className="card__content card__padding"
        >
            <PostDetails
                details={undefined}
            />
            <TextBox>
                Text for blogpost
            </TextBox>
            <Connect(Like)
                postId={4}
            />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="wrapper card radius shadowDepth1"
>
      <div>
            <Link
                  replace={false}
                  to="localhost:9090/posts/4"
            >
                  <Image
                        image={
                              Object {
                                    "alt": "Who cares",
                                    "src": "/pictures/picture1.jpg",
                                    "style": Object {
                                      "height": 200,
                                      "width": 200,
                                    },
                                  }
                        }
                  />
            </Link>
            <div
                  className="card__content card__padding"
            >
                  <PostDetails
                        details={undefined}
                  />
                  <TextBox>
                        Text for blogpost
                  </TextBox>
                  <Connect(Like)
                        postId={4}
                  />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        post={
                Object {
                        "id": 4,
                        "image": Object {
                          "alt": "Who cares",
                          "src": "/pictures/picture1.jpg",
                          "style": Object {
                            "height": 200,
                            "width": 200,
                          },
                        },
                        "postUrl": "localhost:9090/posts/4",
                        "text": "Text for blogpost",
                      }
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "post": Object {
            "id": 4,
            "image": Object {
              "alt": "Who cares",
              "src": "/pictures/picture1.jpg",
              "style": Object {
                "height": 200,
                "width": 200,
              },
            },
            "postUrl": "localhost:9090/posts/4",
            "text": "Text for blogpost",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="wrapper card radius shadowDepth1"
>
          <div>
                    <Link
                              replace={false}
                              to="localhost:9090/posts/4"
                    >
                              <Image
                                        image={
                                                  Object {
                                                            "alt": "Who cares",
                                                            "src": "/pictures/picture1.jpg",
                                                            "style": Object {
                                                              "height": 200,
                                                              "width": 200,
                                                            },
                                                          }
                                        }
                              />
                    </Link>
                    <div
                              className="card__content card__padding"
                    >
                              <PostDetails
                                        details={undefined}
                              />
                              <TextBox>
                                        Text for blogpost
                              </TextBox>
                              <Connect(Like)
                                        postId={4}
                              />
                    </div>
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="wrapper card radius shadowDepth1"
>
          <div>
                    <Link
                              replace={false}
                              to="localhost:9090/posts/4"
                    >
                              <Image
                                        image={
                                                  Object {
                                                            "alt": "Who cares",
                                                            "src": "/pictures/picture1.jpg",
                                                            "style": Object {
                                                              "height": 200,
                                                              "width": 200,
                                                            },
                                                          }
                                        }
                              />
                    </Link>
                    <div
                              className="card__content card__padding"
                    >
                              <PostDetails
                                        details={undefined}
                              />
                              <TextBox>
                                        Text for blogpost
                              </TextBox>
                              <Connect(Like)
                                        postId={4}
                              />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    post={
        Object {
            "id": 4,
            "image": Object {
              "alt": "Who cares",
              "src": "/pictures/picture1.jpg",
              "style": Object {
                "height": 200,
                "width": 200,
              },
            },
            "postUrl": "localhost:9090/posts/4",
            "text": "Text for blogpost",
          }
    }
/>,
}
`;
